rust/opendp-ffi/src/util.rs

```diff
@@ -325,22 +326,25 @@ mod tests {
 
     #[test]
     fn test_type_args_try_from_vec() {
-        let parsed: TypeArgs = "<Vec<i32>>".try_into().unwrap_test();
-        let explicit = TypeArgs(vec![Type::of::<Vec<i32>>()]);
+        let temp = "<Vec<i32>>".to_string();
+        let parsed: Vec<Type> = parse_type_args(temp.as_ptr() as *const c_char, 1).unwrap_test();
```

Comment:
Thanks. I did get a segfault, this commit talks about it: https://github.com/opendp/opendp/pull/85/commits/b3d4b5500a46ce9ec0b441cbbb0e125d26da9690

I thought I fixed it by switching to a String. But String::as_ptr doesn't add the null either. I would guess the only reason this passed was because this was on the heap, and the memory region after the pointer happened to be zeroed? I see that CString adds a null at the end.

Reply:
Yeah, it must've been luck that there were nulls at the right places. That's the main purpose of CStr/CString, to make sure that things are laid out the way C expects.